src/App.tsx

```diff
@@ -15,47 +16,126 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+// Use this function in the render method to prepare goods
+export function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) {
+  // To avoid the original array mutation
+  const visibleGoods = [...goods];
+
+  // Sort and reverse goods if needed
+  // eslint-disable-next-line no-console
+  console.log(sortType, isReversed);
+
+  if (sortType) {
+    visibleGoods.sort((g1, g2) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return g1.localeCompare(g2);
+
+        case SortType.LENGTH:
+          return g1.length - g2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSort] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const reverse = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const sortAlphabatically = () => {
+    setSort(SortType.ALPHABET);
+  };
+
+  const sortByLength = () => {
+    setSort(SortType.LENGTH);
+  };
+
+  const reset = () => {
+    setIsReversed(false);
+    setSort(SortType.NONE);
+  };
+
+  const visibleGoods = getReorderedGoods(goodsFromServer, sortType, isReversed);
+  const resetCheck = isReversed || sortType !== SortType.NONE;
```

Comment:
it is a boolean variable, so this should have a boolean name

Reply:
A name just came to my mind which should be better, I will use that.